java/dataset/src/main/java/org/apache/arrow/dataset/file/JniWrapper.java

```diff
@@ -45,6 +45,17 @@ private JniWrapper() {
    */
   public native long makeFileSystemDatasetFactory(String uri, int fileFormat);
 
+  /**
+   * Create FileSystemDatasetFactory and return its native pointer. The pointer is pointing to a
+   * intermediate shared_ptr of the factory instance.
+   *
+   * @param uris List of file uris to read, each path pointing to an individual file
```

Comment:
Is it working only for specific individual file or it is also working for a directory path that contains many individual files?

Reply:
It will only work for specific individual files, and confirmed this by testing locally. I believe this matches the behaviour of the arrow C++ libraries. We would need to use a selector in order to crawl a directory.

Reply:
Is there any purpose in keeping the `String, int` overload if we have `String[], int` which should be strictly more general?

Reply:
The `String, int` overload correctly handles the user passing a directory, which the `String[], int` overload does not (Each `String` needs to be a complete uri to a source). I'll look into whether I could handle both directories and complete URIs in the `String[], int` implementation, in which case we wouldn't need `String, int`.

Reply:
Aha, thank you.